containers/podman-desktop-extension-ai-lab
# feat(Inference): supporting gpu layers

### What does this PR do?

Add an `experimental GPU` preference. When enabled, it will uses the matching container image for the VMType used

- On WSL and with a NVIDIA GPU 
- On Libkrun => llamacpp mac vulkan 

### Screenshot / video of UI

New option `Experimental GPU` in podman desktop preferences

![image](https://github.com/containers/podman-desktop-extension-ai-lab/assets/42176370/01c7d1d9-aa7b-42ea-8d45-f6a1ac5e7408)

### Side by Side playground comparison 

https://github.com/containers/podman-desktop-extension-ai-lab/assets/42176370/6a8c5308-dc96-40bc-9200-210bd1fe9477

### What issues does this PR fix or reference?

Need https://github.com/containers/podman-desktop-extension-ai-lab/pull/1278 to be merged
Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1279

### How to test this PR?

**Require to be on windows with a WSL podman machine and a NVIDIA GPU**

- Turn on the experimental flag in the preference
- Set a value for the gpu offloading (E.g. 20)
- Start the inference server
- Go to playground, notice the speed
